sw/host/opentitanlib/src/proxy/protocol.rs

```diff
@@ -0,0 +1,162 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::{Deserialize, Serialize};
+
+use crate::io::gpio::{PinMode, PullMode};
+use crate::io::spi::TransferMode;
+use crate::transport::TransportError;
+use crate::util::voltage::Voltage;
+
+#[derive(Serialize, Deserialize)]
+pub enum Message {
+    Req(Request),
+    Res(Result<Response, TransportError>),
+}
+
+#[derive(Serialize, Deserialize)]
+pub enum Request {
+    Gpio { id: String, command: GpioRequest },
```

Comment:
This file was part of the "prerequisite commit" that I had included in the initial PR because GitHub has no support for a chain of dependent PRs being reviewed concurrently.  The file has now been removed from this PR, as the other PR has merged.